pinot-common/src/main/java/org/apache/pinot/common/function/scalar/comparison/PolymorphicComparisonScalarFunction.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar.comparison;
+
+import javax.annotation.Nullable;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SameOperandTypeChecker;
+import org.apache.pinot.common.function.FunctionInfo;
+import org.apache.pinot.common.function.PinotScalarFunction;
+import org.apache.pinot.common.function.sql.PinotSqlFunction;
+import org.apache.pinot.common.utils.DataSchema;
+
+
+/**
+ * Base class for polymorphic comparison scalar functions
+ */
+public abstract class PolymorphicComparisonScalarFunction implements PinotScalarFunction {
+
+  protected static final double DOUBLE_COMPARISON_TOLERANCE = 1e-7d;
+
+  @Nullable
+  @Override
+  public PinotSqlFunction toPinotSqlFunction() {
+    return new PinotSqlFunction(getName(), ReturnTypes.BOOLEAN_FORCE_NULLABLE, new SameOperandTypeChecker(2));
```

Comment:
That makes sense, I'd implemented this since the `PinotScalarFunction` interface doesn't have a default implementation for the `toPinotSqlFunction` method (and I hadn't noticed that it is annotated as `Nullable` with `null` handled appropriately at the call-site in `PinotOperatorTable`). I've added a default implementation of the method in the interface that returns `null` and removed this overridden implementation.

Reply:
@Jackie-Jiang this change caused a test failure in `NullHandlingIntegrationTest::testNullLiteralSelectionInV2`. The query `SELECT greater_than(null, 1) FROM mytable` fails with:

```
No match found for function signature greater_than(<NULL>, <NUMERIC>)
``` 

So I guess the standard function definition doesn't tolerate nulls. I've reverted the change because in Postgres for instance, `SELECT greater_than(null, 1) FROM mytable` does return `null` and our function implementations can also handle nulls through the `FunctionInvoker`'s null intolerant function handling.

Reply:
I think I know the reason. `greater_than` cannot be used explicitly. Can you try `SELECT null > 1 FROM mytable` and see if it works? The test is testing some non-standard SQL behavior, and I don't know if we want to keep the old behavior. Basically when querying `SELECT null > 1 FROM mytable` it will match the Calcite standard `SqlOperator`; when querying `SELECT greater_than(null, 1) FROM mytable` it will match this custom `SqlOperator` which might have different behavior thus causing confusion.
If we want to keep the support of explicit `greater_than`, we can add it under `PinotOperatorTable.STANDARD_OPERATORS_WITH_ALIASES`. But I doubt if anyone is using it this way.

Reply:
Ah, you're right, this actually isn't about nullability. Even a query like `SELECT greater_than(2, 1) FROM mytable` fails with the error: `No match found for function signature greater_than(<NUMERIC>, <NUMERIC>)` if we remove this overridden method because we wouldn't be registering the scalar function in the operator table anymore. And yeah, `SELECT null > 1 FROM mytable` works just fine.

> If we want to keep the support of explicit greater_than, we can add it under PinotOperatorTable.STANDARD_OPERATORS_WITH_ALIASES. But I doubt if anyone is using it this way.

Makes sense, I don't see any harm in retaining support for this explicit syntax, so I've added aliases for all these comparison operators.